Fixes #33551 - allow searching audits by current_user

ares opened the pull request:
A user would like to define a filter that would allow all users to see
their own audits. The only thing that's missing is to define the
scoped_search on Audit model that would accept the current_user value
for audits' user_id.

This PR allows to search audits by the following terms

```
user.id = 1
user.id = 100
user.id = current_user
```

the syntax is consitent with other places we have similar support (e.g.
the tasks).


<!---

Thank you for contributing to The Foreman, please read the
[following guide](https://www.theforeman.org/contribute.html), in short:

* [Create an issue](https://projects.theforeman.org/projects/foreman/issues)
* Reference the issue via `Fixes #1234` in the commit message
* Prefer present-tense, imperative-style commit messages
* Mark all strings for translation, see [1]
* Suggest prerequisites for testing and testing scenarios following example above.
* Prepend `[WIP]` for work in progress to prevent bots from triggering actions
* Be patient, we will do our best to take a look as soon as we can
* Explain the purpose of the PR, attach screenshots if applicable
* List all prerequisites for testing (e.g. VMware cluster, two smart proxies...)
* Reviewers often use extensive list of items to check, have a look prior submitting [2]
* Be nice and respectful

1: https://projects.theforeman.org/projects/foreman/wiki/Translating#Translating-for-developers
2: https://github.com/theforeman/foreman/blob/develop/developer_docs/pr_review.asciidoc
-->

ares committed:
Fixes #33551 - allow searching audits by current_user

A user would like to define a filter that would allow all users to see
their own audits. The only thing that's missing is to define the
scoped_search on Audit model that would accept the current_user value
for audits' user_id and user.

This PR allows to search audits by the following terms

```
user.id = 1
user.id = 100
user.id = current_user

user = admin
user = current_user
```

the syntax is consitent with other places we have similar support (e.g.
the tasks).

ares commented:
To easily test this without setting up a filter in the default role, you can go to audits page and try searching as described in the commit message. Filters use the same searching.

nbarrientos commented:
This is cool :)

ezr-ondrej reviewed (commented):
I really love this, but wouldn't it be better to have it on `user` asociation instead?
user.id is not where I'd expect `current_user` value :thinking:
Or are we worried that there might be user with `login = current_user`?

ezr-ondrej commented:
ping @ares? I don't feel it's a blocker, but I would like to know if it was intentional, or not :)

ares committed:
Fixes #33724 - disable picking a login with a special meaning

Users can pick their login, however there is a reserved word used in
searching - current_user. This patch prevents user from using this
login (username).

ezr-ondrej commented:
[test foreman]

ezr-ondrej reviewed (approved):
Thanks @ares nice improvement ! :+1:

ares commented:
I'll try to squeeze this into other demo today, thanks for merging.
